oracle/types/market.go

```diff
@@ -1,160 +1,94 @@
-	for tickerStr, config := range marketMap.Providers {
-		ticker, ok := marketMap.Tickers[tickerStr]
-		if !ok {
-			return ProviderMarketMap{}, fmt.Errorf("ticker %s not found in market map", tickerStr)
-		}
-
-		for _, provider := range config.Providers {
-			if provider.Name != name {
+	var (
+		// Track all of the tickers that the provider will be providing data for.
+		providerTickers = make([]ProviderTicker, 0)
+		// Maintain a set of off-chain tickers that have been seen to avoid duplicates.
+		seenOffChainTickers = make(map[string]struct{})
+	)
+
+	// Iterate through every single market and its provider configurations to find the
+	// provider configurations that match the provider name.
+	for _, market := range marketMap.Markets {
+		for _, cfg := range market.ProviderConfigs {
+			if cfg.Name != name {
+				continue
+			}
+			if _, ok := seenOffChainTickers[cfg.OffChainTicker]; ok {
 				continue
 			}
 
-			tickers[ticker] = provider
-			break
-		}
-	}
-
-	return NewProviderMarketMap(name, tickers)
-}
-
-// NewProviderMarketMap returns a new provider market map.
-func NewProviderMarketMap(name string, tickerConfigs TickerToProviderConfig) (ProviderMarketMap, error) {
-	if len(name) == 0 {
-		return ProviderMarketMap{}, fmt.Errorf("provider name cannot be empty")
-	}
-
-	if len(tickerConfigs) == 0 {
-		return ProviderMarketMap{
-			Name:          name,
-			TickerConfigs: make(map[mmtypes.Ticker]mmtypes.ProviderConfig),
-			OffChainMap:   make(map[string]mmtypes.Ticker),
-		}, nil
-	}
-
-	offChainMap := make(map[string]mmtypes.Ticker)
-	for ticker, config := range tickerConfigs {
-		if err := ticker.ValidateBasic(); err != nil {
-			return ProviderMarketMap{}, fmt.Errorf("invalid ticker %s: %w", ticker, err)
-		}
-
-		if err := config.ValidateBasic(); err != nil {
-			return ProviderMarketMap{}, fmt.Errorf("invalid provider config for %s: %w", ticker, err)
-		}
-
-		if config.Name != name {
-			return ProviderMarketMap{}, fmt.Errorf("expected provider config name %s, got %s", name, config.Name)
+			providerTicker := NewProviderTicker(
+				cfg.Name,
+				cfg.OffChainTicker,
+				cfg.Metadata_JSON,
+				DefaultTickerDecimals,
+			)
+			providerTickers = append(providerTickers, providerTicker)
+			seenOffChainTickers[cfg.OffChainTicker] = struct{}{}
 		}
-
-		offChainMap[config.OffChainTicker] = ticker
 	}
 
-	return ProviderMarketMap{
-		Name:          name,
-		TickerConfigs: tickerConfigs,
-		OffChainMap:   offChainMap,
-	}, nil
+	return providerTickers, nil
 }
 
-// GetTickers returns the tickers from the provider market map.
-func (pmm *ProviderMarketMap) GetTickers() []mmtypes.Ticker {
-	tickers := make([]mmtypes.Ticker, 0, len(pmm.TickerConfigs))
-	for ticker := range pmm.TickerConfigs {
-		tickers = append(tickers, ticker)
-	}
-	return tickers
+// SimpleProviderTicker is a simple representation of a provider ticker. Provider's
+// can utilize this to easily configure custom tickers.
+type SimpleProviderTicker struct {
```

Comment:
Would prefer calling this "Base"

Reply:
why do we have SimpleProviderTicker and DefaultPRoviderTIcker?

Reply:
yea im gonna remove the other

Reply:
removed